packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -278,6 +278,26 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
         // done with 'session_token_id' at this point, do not return it.
         delete grant.session_token_id;
 
+        try {
+          let uid = sessionToken.uid;
+
+          // TODO: is it still true that "some grant flows won't have the uid
+          // in `credentials`"? (see lib/routes/utils/oauth.js) Or is it true that
+          // we will "refuse to call oauthdb.grantTokensFromSessionToken without
+          // a valid sessionToken"? (see test/local/routes/oauth.js)
```

Comment:
sessionToken may be null here but if you have one it will have a uid

Reply:
So, yes, both are true. If the request was authed with a sessionToken by the time we get here it has a uid. If it was a `refresh_token` or `authorization_code` it won't